lib/pf/config.pm

```diff
@@ -687,10 +688,16 @@ sub readProfileConfigFile {
                     foreach my $field (qw(locale mandatory_fields sources filter provisioners) ) {
                         $profile->{$field} = [split(/\s*,\s*/, $profile->{$field} || '')];
                     }
-                    if ($profile_id ne 'default') {
-                        #Adding filters in profile order
+                    my $filters = $profile->{'filter'};
+                    if($profile_id ne 'default' && @$filters) {
+                        my @filterObjects;
                         foreach my $filter (@{$profile->{'filter'}}) {
-                            push @Profile_Filters, pf::factory::profile::filter->instantiate($profile_id,$filter);
+                            push @filterObjects, pf::factory::profile::filter->instantiate($profile_id,$filter);
+                        }
+                        if($profile->{filter_match_style} eq 'all') {
```

Comment:
Should be 
if(defined($profile->{filter_match_style}) && $profile->{filter_match_style} eq 'all') 
because when upgrading from a previous version, that parameter won't be there and will generate undefined variable warnings